src/main/haskell/language-kore/src/Data/Kore/ASTPrettyPrint.hs

```diff
@@ -255,7 +255,7 @@ instance PrettyPrint UnifiedSortVariable where
     prettyPrint flags (MetaSortVariable sv)   =
         writeOneFieldStruct flags "MetaSortVariable" sv
 
-instance PrettyPrint UnifiedVariable where
+instance PrettyPrint (UnifiedVariable Variable) where
     prettyPrint flags (ObjectVariable sv) =
         writeOneFieldStruct flags "ObjectVariable" sv
```

Comment:
The Writer monad is also frowned upon. A good alternative would be something like a 'streaming' library. I think conduit and pipes can do the job. This is however a pretty big change and maybe someone else, better versed in Haskell could pitch in with an opinion. Relevant: https://twitter.com/gabrielg439/status/659170544038707201?lang=en, https://www.fpcomplete.com/blog/2017/06/readert-design-pattern.

Not that I don't think this is a big issue, but more opinions would be welcomed.

Reply:
It seems from the links you've sent that a writer monad with a strict tell would be ok. And that's fairly easy to write, and a super easy change. So I would wait until we see some space leaks from it to decide if it's worth the change.